### PR TITLE
API Complete and running. Ready for performance testing.

### PR DESCRIPTION
**Pull request by JakeDavisCodes**

### CODE REVIEW

File: server/controller.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      req.body.question_id = req.params.question_id;
+      model.a.insert(req.body)
+        .then(() => res.sendStatus(201))
+        .catch((err) => {
+          console.error(err);
+          res.status(500).json(err);
+        });
+    },
+    helpful: (req, res) => {
+      if (!req.params.answer_id) res.status(400).json({ error: 'please supply a answer_id' });
```
Comment (BrettEastman) [inline server/controller.js:107]: Your code is overall excellent and I can't see anything wrong with it. If I am being super picky, I would say you should change this error statement to: "Please supply an answer ID". You just forgot the "an" is all. Same with the next one.

File: server/model.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                   FROM questions
+                   WHERE product_id = $1
+                   AND reported = ${false}
+                   ORDER BY question_helpfulness DESC, question_id ASC
+                   LIMIT $2 OFFSET $3`;
+      return client.query(sql, [p_id, count, offset]);
+    },
+    insert: ({
+      product_id, name, email, body,
+    }) => client.query('SELECT MAX(question_id) from "questions"')
```
Comment (BrettEastman) [inline server/model.js]: Couldn't you omit the question ID and have the SERIAL ID advance the ID number for you?

**Review by BrettEastman — COMMENTED** [review]
Great work, Jake! I added two comments that may be worth looking into. But, overall, everything looks excellent!

**Review by kyrianite — APPROVED** [review]
Apologies for the belated review. Looks good though!

**Pull request merged**